datadogreceiver/translator.go

```diff
@@ -0,0 +1,166 @@
+// Copyright 2021, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package datadogreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/datadogreceiver"
+
+import (
+	"encoding/binary"
+	"encoding/json"
+	"mime"
+	"net/http"
+	"strings"
+
+	"runtime/debug"
+
+	datadogpb "github.com/DataDog/datadog-agent/pkg/trace/exportable/pb"
+	"github.com/tinylib/msgp/msgp"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	semconv "go.opentelemetry.io/collector/semconv/v1.6.1"
+)
+
+func addResourceData(req *http.Request, rs *pcommon.Resource) {
+	version := ""
+	bi, ok := debug.ReadBuildInfo()
```

Comment:
Is this the right way to grab the receiver version? Since this should be unchanging, should / can it be extracted to a higher concept else where?

Reply:
You could set a string value in the receiver code that you read here. Then you'd need to update as part of the release process.

eg version.go
```
package datadogreceiver

const version string = "0.1.0"
```